File: milestone 2/python_modules/module03/ex6/ft_data_alchemist.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(void)
{
    std::cout << "=== Game Data Alchemist ===\n" << endl;

    vector <string> players = {
        "Alice", "bob", "Charlie", "dylan", "Emma", "Gregory", "john", "kevin", "Liam"
    };

    std::cout << "Initial list of players: ";
    for (int i = 0; i < players.size(); i++)
    {
        std::cout << players[i];
        if (i < players.size() - 1)
            std::cout << ", ";
    }
    std::cout << endl;
    std::cout << "\nNew list with all names capitalized: ";
    for (int i = 0; i < players.size(); i++)
    {
        if (players[i][0] >= 97 && players[i][0] <= 122)
            players[i][0] -= 32;
        std::cout << players[i];
        if (i < players.size() - 1)
            std::cout << ", ";
    }
    std::cout << endl;
    std::cout << "\nNew list of capitalized names only: ";
    vector <string> cap_only;
    for (int i = 0; i < players.size(); i++)
    {
        if (players[i][0] >= 65 && players[i][0] <= 90)
            cap_only.push_back(players[i]);
    }
    for (int i = 0; i < cap_only.size(); i++)
    {
        
    }
    std::cout << endl;
}

File: milestone 2/python_modules/module03/ex6/ft_data_alchemist.py
import random

if __name__ == "__main__":
    print("=== Game Data Alchemist ===\n")

    players: list = ['Alice', 'bob', 'Charlie', 'dylan', 'Emma', 'Gregory', 'john', 'kevin', 'Liam']
    print(f"Initial list of players: {players}")

    all_capitalized: list = [name.capitalize() for name in players]
    print(f"New list with all names capitalized: {all_capitalized}")

    only_capitalized: list = [name for name in players if name[0].isupper()]
    print(f"New list of capitalized names only: {only_capitalized}")

    score_dict: dict = {name: random.randint(0, 1000) for name in all_capitalized}
    print(f"Score dict: {score_dict}")

    average: float = round(sum(score_dict.values()) / len(score_dict), 2)
    print(f"Score average is {average}")

    high_scores: dict = {name: score for name, score in score_dict.items() if score > average}
    print(f"High scores: {high_scores}")